MitoMathHelper.js

```diff
@@ -172,4 +172,94 @@ const MitoMathHelper = class MitoMathHelper {
         let normalizedDistance = aToPDoToAToB / aToBSquared;
         return [linePointA[0] + (aToB[0] * normalizedDistance), linePointA[1] + (aToB[1] * normalizedDistance)]
     }
+
+    /**
+     * returns magnitude of a motion
+     * @param vector
+     * @returns {number}
+     * @private
+     */
+    static _getMagnitude(vector) {
+        return Math.hypot(vector[0], vector[1]);
+    }
+
+    /**
+     * returns magnitude squared
+     * @param vector
+     * @returns {number}
+     * @private
+     */
+    static _getMagnitudeSquared(vector) {
+        return vector[0] * vector[0] + vector[1] * vector[1];
+    }
+
+    /**
+     * returns dot product
```

Comment:
stay consistent with how we comment everything else or documentation only, comment comments dont matter

```
Returns the dot product.
```

Reply:
its fine to keep this function IMO because dot product is not very intuitive whats going on